completion_aggregator/api/v0/views.py

```diff
@@ -0,0 +1,344 @@
+          completable sections.
+
+        Example response:
+
+            GET /api/course
+
+            {
+              "pagination": {
+                "count": 14,
+                "page": 1,
+                "next": "/api/completion/v0/course/?page=2,
+                "previous": None
+              },
+              "results": [
+                {
+                  "course_key": "edX/DemoX/Demo_course",
+                  "completion": {
+                    "earned": 42.0,
+                    "possible": 54.0,
+                    "ratio": 0.77777777777778
+                  },
+                  "chapter": [
+                    {
+                      "course_key": "edX/DemoX/Demo_course",
+                      "block_key": "i4x://edX/DemoX/chapter/chapter1",
+                      "completion": {
+                        "earned: 20.0,
+                        "possible": 30.0,
+                        "ratio": 0.6666666666667
+                      }
+                    },
+                    {
+                      "course_key": "edX/DemoX/Demo_course",
+                      "block_key": "i4x://edX/DemoX/chapter/chapter2",
+                      "completion": {
+                        "earned: 22.0,
+                        "possible": 24.0,
+                        "ratio": 0.9166666666667
+                      }
+                    }
+                  ]
+                },
+                {
+                  "course_key": "course-v1:GeorgetownX+HUMX421-02x+1T2016",
+                  "completion": {
+                    "earned": 12.0,
+                    "possible": 24.0,
+                    "ratio": 0.5
+                  },
+                  "chapter": [
+                    {
+                      "course_key": "course-v1:GeorgetownX+HUMX421-02x+1T2016",
+                      "block_key": "block-v1:GeorgetownX+HUMX421-02x+1T2016+type@chapter+block@Week-2-TheVitaNuova",
+                      "completion": {
+                        "earned: 12.0,
+                        "possible": 24.0,
+                        "ratio": 0.5
+                      }
+                    }
+                  ]
+                }
+              ]
+            }
+
+    """
+
+    def get(self, request):
+        """
+        Handler for GET requests.
+        """
+        paginator = self.pagination_class()  # pylint: disable=not-callable
+        mobile_only = (self.request.query_params.get('mobile_only', 'false')).lower() == 'true'
+
+        # Paginate the list of active enrollments, annotated (manually) with a student progress object.
+        enrollments = UserEnrollments(self.user).get_enrollments()
+
+        if mobile_only:
+            course_keys = []
+            for course_enrollment in enrollments:
+                course_keys.append(course_enrollment.course_id)
+            course_overview_list = CourseOverview.objects.filter(id__in=course_keys, mobile_available=True)
```

Comment:
@jcdyer how can we migrate this? shall we just remove this option of mobile_only?

Reply:
No.  IIRC, mobile_only is something mckinsey asked for.  The way we've been dealing with dependencies to edx-platform is to put them into a function in completion_aggregator/compat.py (`get_mobile_only_courses()`, for example), that takes and returns CourseKey objects, and then add a stub implementation in `test_utils/compat.py` that makes it work properly for tests.  The connection to edx platform will not be covered by the unittests tests, but we can test it manually, and add integration tests to edx-platform.

I'll add a ticket to include similar functionality in the v1 api, and to write integration tests for the mobile_available flag on v0 and v1.

Reply:
Or you can update compat.get_enrollments to take a mobile_only flag.

But I may have misunderstood your question.

Reply:
> I'll add a ticket to include similar functionality in the v1 api, and to write integration tests for the mobile_available flag on v0 and v1.

I can work on this on this sprint if you create it, or I can create it myself too.